x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/Engine.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.entsearch.engine;
+
+import org.elasticsearch.ElasticsearchParseException;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.io.stream.ReleasableBytesStreamOutput;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.util.BigArrays;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.core.Tuple;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentFactory;
+import org.elasticsearch.xcontent.XContentParser;
+import org.elasticsearch.xcontent.XContentParserConfiguration;
+import org.elasticsearch.xcontent.XContentType;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.elasticsearch.xcontent.ConstructingObjectParser.constructorArg;
+
+public class Engine implements Writeable, ToXContentObject {
+    private final String name;
+    private final String[] indices;
+
+    /**
+     * Public constructor.
+     *
+     * @param name The name of the engine.
+     * @param indices The list of indices targeted by this engine.
+     */
+    public Engine(String name, String[] indices) {
+        this.name = name;
+        this.indices = indices;
+    }
+
+    public Engine(StreamInput in) throws IOException {
+        this.name = in.readString();
```

Comment:
Do we need to do any validation here?

Reply:
What kind of validation are you thinking? That the name contains alpha-numeric chars only?

Reply:
In the Rails application we do the following validation: 

1. The name must consist of only alphanumerics and hyphens, and must be unique (The REST API would validate uniqueness)
2. We had been doing some validation requiring indices be present and non-empty, but it looks like we may have removed that.

Reply:
++, we probably want to check the indices exist

Reply:
@carlosdelest I made the assumption that checking for the existence of indices would be part of the API contract, along with name uniqueness on POST Create calls. WDYT?

Reply:
+1 @kderusso, the validation should happen at the API level.